src/main.go

```diff
@@ -0,0 +1,124 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
+	tcollama "github.com/testcontainers/testcontainers-go/modules/ollama"
+	"io"
+	"log"
+	"net/http"
+	"os"
+	"regexp"
+)
+
+type TagResponse struct {
+	Models []struct {
+		Name string `json:"name"`
+	} `json:"models"`
+}
+
+type GenerateRequest struct {
+	Model  string `json:"model"`
+	Stream bool   `json:"stream"`
+	Prompt string `json:"prompt"`
+}
+
+func startContainer(image string) (string, error) {
+	ctx := context.Background()
+	ollamaContainer, err := tcollama.Run(ctx, image, testcontainers.CustomizeRequestOption(func(req *testcontainers.GenericContainerRequest) error {
+		req.Name = normalizeString(image)
+		req.Reuse = true
+		return nil
+	}))
+	if err != nil {
+		log.Fatalf("failed to start container: %s", err)
+		return "", err
+	}
+	return ollamaContainer.Endpoint(ctx, "http")
+}
+
+func normalizeString(input string) string {
+	// Define a regular expression that matches any character that is not alphanumeric, _, or -
+	re := regexp.MustCompile(`[^a-zA-Z0-9_-]`)
```

Comment:
nit: usually I'd move these as "global"/"module" scoped var at the top of the file (I kinda assumed that's why `MustCompile` exists because if you'd do it in a function, you could check the error and use the more safe `regex.Compile`)